views/js/runner/plugins/content/dialog/dialog.js

```diff
@@ -27,35 +27,117 @@ define([
     'use strict';
 
     /**
+     * The public name of the plugin
+     * @type {String}
+     */
+    var pluginName = 'dialog';
+
+    /**
+     * The prefix of actions triggered through the event loop
+     * @type {String}
+     */
+    var actionPrefix = 'tool-' + pluginName + '-';
+
+    /**
      * Returns the configured plugin
      */
     return pluginFactory({
-        name : 'dialog',
+        name : pluginName,
 
         /**
          * Initialize the plugin (called during runner's init)
          */
         init : function init(){
             var testRunner = this.getTestRunner();
+            var testData = testRunner.getTestData() || {};
+            var testConfig = testData.config || {};
+            var pluginShortcuts = (testConfig.shortcuts || {})[pluginName] || {};
             var alerts = [];
             var confirms = [];
+            var opened = [];
+            var dialogShortcut = shortcutRegistry($('body'), {
+                propagate: false,
+                prevent: true
+            });
+
+            /**
+             * Closes a dialog with accept
+             * @param {dialog} dialog - The instance of the dialog
+             */
+            function closeAccept(dialog) {
+                // TODO: improve the dialog implementation in order to provide a better API
+                dialog.trigger('okbtn.modal').hide();
+            }
+
+            /**
+             * Closes a dialog with rejection
+             * @param {dialog} dialog - The instance of the dialog
+             */
+            function closeReject(dialog) {
+                dialog.hide();
+            }
+
+            /**
+             * Closes the last opened dialog
+             * @param {Boolean} accept Whether the dialog should be accepted or not
+             */
+            function closeLast(accept) {
+                var handle = opened.length && opened[opened.length - 1];
+                if (handle) {
+                    if (accept) {
+                        closeAccept(handle.dialog);
+                    } else {
+                        closeReject(handle.dialog);
+                    }
+                }
+            }
 
+            /**
+             * Add dialog on top of the provided stack
+             * @param {String} namespace - The event namespace that scope the dialog
+             * @param {Array} stack - The dialogs stack on which push the new instance
+             * @param {dialog} dialog - The instance of the dialog
+             */
             function addHandle(namespace, stack, dialog) {
-                stack.push({
+                var handle = {
                     context: namespace,
                     dialog: dialog
-                });
+                };
+
+                // prevents all registered shortcuts to be triggered
+                // and brings back the dialog shortcuts
+                shortcut.disable();
```

Comment:
maybe this could be named something like globalShortcuts or, even better, windowShortcuts if those are bound to window ?